### PR TITLE
[TIMOB-23864] CLI: Build process does not exit if using an incompatib…

### PR DESCRIPTION
**Pull request by feons**
…le module based on apiversion

https://jira.appcelerator.org/browse/TIMOB-23864

A PR for TiSDK will be needed once this is merged.

### CODE REVIEW

File: lib/timodule.js [hunk 1]
```diff
@@ -179,13 +179,13 @@ function detect(paramsOrSearchPaths, logger, callback, bypassCache) {
...
  * @param {Array<String>|String} [params.deployType] - An array of deploy types or a string of comma-separated deploy types to filter by.
- * @param {String} [params.sdkVersion] - The version of the Titanium SDK to be used for the minimum SDK check.
+ * @param {Object} [params.tiManifest] - The Titanium SDK manifest data.
  * @param {Array<String>} [params.searchPaths] - An array of paths to search for Titanium modules.
  * @param {Object} [params.logger] - A logger instance.
  * @param {Function} [params.callback] - A function to call when done.
  * @param {Boolean} [params.bypassCache=false] - When true, re-detects all modules.
  */
-function find(modulesOrParams, platforms, deployType, sdkVersion, searchPaths, logger, callback, bypassCache) {
+function find(modulesOrParams, platforms, deployType, tiManifest, searchPaths, logger, callback, bypassCache) {
```
Comment (cb1kenobi) [inline lib/timodule.js]: Just to be clear, technically you're passing in both an `sdkVersion` and a `moduleAPIVersion`, but due to backward compatibility you pass in via an object `tiManifest`?

Comment (feons) [inline lib/timodule.js]: Yes.

File: lib/timodule.js [hunk 2]
```diff
@@ -287,6 +296,15 @@ function find(modulesOrParams, platforms, deployType, sdkVersion, searchPaths, l
 								return;
 							}
 
+							if (moduleAPIVersion && moduleAPIVersion[platform] && info.manifest && info.manifest.apiversion && ~~info.manifest.apiversion < ~~moduleAPIVersion[platform]) {
```
Comment (cb1kenobi) [inline lib/timodule.js]: I don't think we can do a less than comparison. For example, I highly doubt that Android supports v1, v2, and v3 modules.

**Comment by cb1kenobi** [comment]
A little difficult to test, but it passes CR. Will FR when the TiSDK changes land. **APPROVED**.

**Pull request merged**